src/core/algorithms/statistics/data_stats.cpp

```diff
@@ -505,13 +487,122 @@ Statistic DataStats::GetMedianAD(size_t index) const {
     return Statistic(median_ad, &double_type, false);
 }
 
-Statistic DataStats::GetNumNulls(size_t index) const {
-    if (all_stats_[index].num_nulls.HasValue()) return all_stats_[index].num_nulls;
+Statistic DataStats::GetVocab(size_t index) const {
+    if (all_stats_[index].vocab.HasValue()) return all_stats_[index].vocab;
+    mo::TypedColumnData const& col = col_data_[index];
+    if (col.GetTypeId() != +mo::TypeId::kString) return {};
+
+    mo::StringType string_type;
+    std::string string_data;
+    std::set<char> vocab;
+
+    for (size_t i = 0; i < col.GetNumRows(); i++) {
+        if (col.IsNullOrEmpty(i)) continue;
+        auto const& string_data = mo::Type::GetValue<std::string>(col.GetValue(i));
+        vocab.insert(string_data.begin(), string_data.end());
+    }
+    std::string temp(vocab.begin(), vocab.end());
+
+    std::byte const* res = string_type.MakeValue(temp);
+
+    return Statistic(res, &string_type, false);
```

Comment:
But we don't actually initialize members of Statistic object with Type pointer passed, the new object of the same Type we pass is being created whenever we call a ctor.
If we just return string, so we should return an empty one in case column isn't string and it is kinda weird

Reply:
You're right